pkg/engine/armresources.go

```diff
@@ -47,10 +47,9 @@ func GenerateARMResources(cs *api.ContainerService) []interface{} {
 		armResources = append(armResources, userAssignedID, msiRoleAssignment)
 	}
 
-	if !cs.Properties.OrchestratorProfile.IsPrivateCluster() &&
+	if cs.Properties.OrchestratorProfile.KubernetesConfig.LoadBalancerSku == api.StandardLoadBalancerSku &&
 		!isHostedMaster &&
-		!cs.Properties.AnyAgentHasLoadBalancerBackendAddressPoolIDs() &&
-		cs.Properties.OrchestratorProfile.KubernetesConfig.LoadBalancerSku == api.StandardLoadBalancerSku {
+		!cs.Properties.AnyAgentHasLoadBalancerBackendAddressPoolIDs() {
 		isForMaster := false
 		publicIPAddress := CreatePublicIPAddress(isForMaster)
```

Comment:
You obviously didn't change this, but when ever I see bool flags passed into functions, it just screams for two functions. One for creating a pub IP for a master and another for an agent pool. I think this is indicated by the desire to name the bool being passed into `CreatePublicIPAddress`.

I think the oddest part of this function and others like it is that if you called it twice and add the resources  to the template, you would have overlapping resources (two resources with the same name). It just feels odd and opaque.

Reply:
Right. To be clear the state of this PR is "sort of working prototype". Once we get to "definitely working prototype" then it will be appropriate to do some of your suggested code interface improvements.

Reply:
I'll refrain. Just perusing and venting as it's top of mind.

Reply:
Please don't refrain, these comments are being valuably added to the public record! 😋